File: burstLibrary/src/main/java/com/harry1453/burst/explorer/service/GsonDeserializerService.java
package com.harry1453.burst.explorer.service;

import com.google.gson.Gson;

import io.reactivex.Single;

public final class GsonDeserializerService implements DeserializerService {

    private final Gson gson = new Gson();

    @Override
    public <T> Single<T> deserializeObject(String objectData, Class<T> objectClass) {
        return Single.fromCallable(() -> gson.fromJson(objectData, objectClass));
    }
}

File: burstLibrary/src/main/java/com/harry1453/burst/explorer/service/GsonDeserializerService.kt
package com.harry1453.burst.explorer.service

import com.google.gson.Gson

import io.reactivex.Single

class GsonDeserializerService : DeserializerService {

    private val gson = Gson()

    override fun <T> deserializeObject(objectData: String, objectClass: Class<T>): Single<T> {
        return Single.fromCallable { gson.fromJson(objectData, objectClass) }
    }
}
